tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -419,11 +419,7 @@ pub struct Format<F = Full, T = SystemTime> {
 // === impl Writer ===
 
 impl<'writer> Writer<'writer> {
-    // TODO(eliza): consider making this a public API?
-    // We may not want to do that if we choose to expose specialized
-    // constructors instead (e.g. `from_string` that stores whether the string
-    // is empty...?)
-    pub(crate) fn new(writer: &'writer mut impl fmt::Write) -> Self {
+    pub fn new(writer: &'writer mut impl fmt::Write) -> Self {
```

Comment:
If we're going to make this a public API, it should have API documentation. 

Also, since it's a constructor, let's add
```suggestion
    #[must_use]
    pub fn new(writer: &'writer mut impl fmt::Write) -> Self {
```
to ensure that the returned value is not accidentally discarded.

Reply:
How does one add the documentation? I can do this if no one else has the time. I don't know too much about how this process works (just doc comments?), or whatever second and third degree affects might need to be documented.